docker/Dockerfile

```diff
@@ -1,45 +1,125 @@
-FROM clux/muslrust:1.70.0 as builder
-WORKDIR /app
-ARG CARGO_BUILD_TARGET=x86_64-unknown-linux-musl
-
-# comma-seperated list of features to enable
+ARG RUST_VERSION=1.71.0
+ARG ALPINE_VERSION=3.18
 ARG CARGO_BUILD_FEATURES=default
+ARG RUST_RELEASE_MODE=debug
+ARG UID=911
+ARG GID=911
+
+####################################################################################################
+## AMD64 builder base
+####################################################################################################
+FROM --platform=${BUILDPLATFORM} blackdex/rust-musl:x86_64-musl-stable-${RUST_VERSION}-openssl3 AS base-amd64
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV CARGO_HOME=/root/.cargo
+ENV PQ_LIB_DIR=/usr/local/musl/pq15/lib
+
+RUN apt update && apt install -y \
+    --no-install-recommends \
+    git
+
+RUN mkdir -pv "${CARGO_HOME}" && \
+    rustup set profile minimal && \
+    rustup target add x86_64-unknown-linux-musl
+
+####################################################################################################
+## ARM64 builder base
```

Comment:
Could you use simple comments, IE `# ARM64 builder base` for these. These don't wrap correctly on the device I'm using.